framework/src/util/SharedLibrary.cpp

```diff
@@ -69,7 +68,14 @@ SharedLibrary::SharedLibrary()
   : d(new SharedLibraryPrivate)
 {}
 
-SharedLibrary::SharedLibrary(const SharedLibrary&) = default;
+SharedLibrary::SharedLibrary(const SharedLibrary& other) : d(new SharedLibraryPrivate) {
```

Comment:
Just use the copy constructor of `SharedLibraryPrivate` instead of doing it manually below.

If the copied raw `m_Handle` pointer is not null, set it to `nullptr` and call `Load()`. Same in the assignment operator. This will ensure `Unload` can be called on a copied `SharedLibrary` instance without messing up the internal library reference counting.

Reply:
I believe that the handle should remain a std::shared_ptr. It removes the necessity to call Load() on the newly constructed object if creating another SharedLibrary object through the copy constructor. Not to mention, calling Load() on the same dll twice doesn't appear to be a good idea. If you would really like me to switch the handle back to a raw pointer and unnecessarily call Load() again, I can, but I'd prefer not to for the reasons above.

Reply:
Yep, but due to my comment above to your other post, unless you absolutely want to keep the handle as a raw pointer, then I won't be doing the extra Load() call in this PR.